kernel/Base/Element.h

```diff
@@ -525,9 +530,19 @@ class Element final : public Geometry::ElementGeometry, public ElementData {
     /// \param owned Whether this element is owned or not by the current
     /// processor.
     void setOwnedByCurrentProcessor(std::size_t owner, bool owned);
-    bool isOwnedByCurrentProcessor() const;
+    bool isOwnedByCurrentProcessor() const final;
     std::size_t getOwner() const;
 
+    const Element* getOwningElement() const final { return this; }
+
+    void visitEntity(MeshEntityVisitor<>& visitor) final {
+        visitor.visit(*this);
+    }
+
+    void visitEntity(MeshEntityVisitor<true>& visitor) const final {
```

Comment:
the `true` is probably the read/write?

Reply:
Const/non-const with respect to the entity, but I have now improved it so that you have `MeshEntityVisitor` as the non const one and `ConstMeshEntityVisitor` as the const one. Though at the cost of having the sligthly worse named `MeshEntityVisitorT` as base for them both.

Reply:
Good choice